rust/worker/src/execution/orchestration/hnsw.rs

```diff
@@ -277,18 +279,60 @@ impl HnswQueryOrchestrator {
                 }
             },
         };
-
         println!("Created HNSW Segment Reader: {:?}", hnsw_segment_reader);
 
+        let record_segment = self
```

Comment:
we should not read the record segment in the request thread i think - everywhere else we push it into operators

Reply:
Good point!